src/Altinn.AccessManagement.Core/Helpers/RightsHelper.cs

```diff
@@ -53,16 +53,106 @@ public static List<AttributeMatch> GetDelegationSubjectAttributeMatches(Delegati
             {
                 To = new List<AttributeMatch> { new AttributeMatch { Id = AltinnXacmlConstants.MatchAttributeIdentifiers.UserAttribute, Value = userId.ToString() } },
                 From = new List<AttributeMatch> { new AttributeMatch { Id = AltinnXacmlConstants.MatchAttributeIdentifiers.PartyAttribute, Value = fromPartyId.ToString() } },
                 Resource = new AttributeMatch { Id = AltinnXacmlConstants.MatchAttributeIdentifiers.ResourceRegistryAttribute, Value = resourceRegistryId }.SingleToList()
             };
         }
+
+        /// <summary>
+        /// Analyses a Right model for a reason for the rights delegation access status
+        /// </summary>
+        public static List<Detail> AnalyzeDelegationAccessReason(Right right)
+        {
+            List<Detail> reasons = new();
+
+            // Analyse why able to delegate
+            if (right.CanDelegate.HasValue && right.CanDelegate.Value)
+            {
+                // Analyze for role access
+                List<RightSource> roleAccessSources = right.RightSources.Where(rs => rs.RightSourceType != Enums.RightSourceType.DelegationPolicy && rs.CanDelegate.HasValue && rs.CanDelegate.Value).ToList();
+                if (roleAccessSources.Any())
+                {
+                    string requiredRoles = string.Join(", ", roleAccessSources.SelectMany(roleAccessSource => roleAccessSource.PolicySubjects.SelectMany(policySubjects => policySubjects)));
+
+                    reasons.Add(new Detail
+                    {
+                        Code = "RoleAccess",
+                        Description = $"Delegator have access through having one of the following role(s) for the reportee party: {requiredRoles}. Note: if the user is a Main Administrator (HADM) the user might not have direct access to the role other than for delegation purposes.",
+                        Params = new Dictionary<string, string>() { { "RoleRequirementsMatches", $"{requiredRoles}" } }
+                    });
+                }
+
+                // Analyze for delegation policy access
+                List<RightSource> delegationPolicySources = right.RightSources.Where(rs => rs.RightSourceType == Enums.RightSourceType.DelegationPolicy && rs.CanDelegate.HasValue && rs.CanDelegate.Value).ToList();
+                if (delegationPolicySources.Any())
+                {
+                    string delegationRecipients = string.Join(", ", delegationPolicySources.SelectMany(delegationPolicySource => delegationPolicySource.PolicySubjects.SelectMany(policySubjects => policySubjects)));
+
+                    reasons.Add(new Detail
+                    {
+                        Code = "DelegationAccess",
+                        Description = $"The user have access through delegation(s) of the right to the following recipient(s): {delegationRecipients}",
+                        Params = new Dictionary<string, string>() { { "DelegationRecipients", $"{delegationRecipients}" } }
```

Comment:
params is protected name
```suggestion
                        DetailParams = new Dictionary<string, string>() { { "DelegationRecipients", $"{delegationRecipients}" } }
```